tds_test.go

```diff
@@ -694,6 +694,62 @@ func TestBadCredentials(t *testing.T) {
 	_ = testConnectionBad(t, params.URL().String())
 }
 
+func TestLeakedConnections(t *testing.T) {
+	goodParams := testConnParams(t)
+	badParams := testConnParams(t)
+	badParams.Database = "unknown_db"
+
+	// Connecting with good credentials should not fail
+	goodConn, err := sql.Open("sqlserver", goodParams.URL().String())
+	if err != nil {
+		t.Fatal("Open connection failed:", err.Error())
+	}
+	err = goodConn.Ping()
+	if err != nil {
+		t.Fatal("Ping with good credentials should not fail, but got error:", err.Error())
+	}
+
+	var localNetAddr sql.NullString
+	err = goodConn.QueryRow("SELECT local_net_address FROM sys.dm_exec_connections WHERE session_id=@@SPID").Scan(&localNetAddr)
+	if err != nil {
+		t.Fatal("cannot scan local_net_address value", err)
+	}
+	if !localNetAddr.Valid {
```

Comment:
this is not correct. local_net_address can be null particularly for non-tcp local connections.